sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -846,12 +846,13 @@ class Analyzer(
       // ResolveReferences. Attributes in the output will be resolved by ResolveGenerate.
       case g @ Generate(generator, _, _, _, _, _) if generator.resolved => g
 
-      case g @ Generate(generator, join, outer, qualifier, output, child) =>
+      case g @ Generate(generator, requiredChildOutput, outer, qualifier, output, child) =>
         val newG = resolveExpression(generator, child, throws = true)
         if (newG.fastEquals(generator)) {
           g
         } else {
-          Generate(newG.asInstanceOf[Generator], join, outer, qualifier, output, child)
+          Generate(newG.asInstanceOf[Generator], child.output, outer,
```

Comment:
we should reserve `requiredChildOutput` here.

Reply:
agree